src/Sulu/Bundle/AdminBundle/FormMetadata/FormXmlLoader.php

```diff
@@ -73,6 +107,36 @@ protected function parse($resource, \DOMXPath $xpath, $type): FormMetadata
         }
         $form->burnProperties();
 
+        $formMetadataCollection = new LocalizedFormMetadataCollection();
+        foreach ($this->locales as $locale) {
+            $formMetadataCollection->add($locale, $this->mapFormsMetadata($form, $locale));
+        }
+
+        return $formMetadataCollection;
+    }
+
+    /**
+     * @param $formMetadata
+     * @param $locale
+     *
+     * @return FormMetadata
+     *
+     * @throws \Exception
+     */
+    private function mapFormsMetadata($formMetadata, $locale)
+    {
+        $form = new FormMetadata();
+        $this->formMetadataMapper->mapChildren($formMetadata->getChildren(), $form, $locale);
```

Comment:
Instead of passing the `FormMetadata` to the `mapChildren` function it would be cleaner to let it return the children as array, and adding the children to the `FormMetadata` could be done in here then. Would make the function more reusable, and similar to how the `schema` works in the next few lines.